### ISSUE TITLE
Sending a magic link will sometimes fail due to password.Generate generating an invalid password

### ISSUE THREAD

**Issue by klajdi369** [supabase/auth#1761]
# Bug report

- [Y] I confirm this is a bug with Supabase, not with my own application. 
- [Y] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Sending a magic link will sometimes fail due to `password.Generate` generating an invalid password that complies with `checkPasswordStrength`

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

Using Selfhosted Docker
1. `GOTRUE_PASSWORD_REQUIRED_CHARACTERS: "!@#$%^&*()-_=+<>"` to this value in .env under auth
2. Sign up a new user
3. Send magic link until it fails
## Expected behavior

The action should sucessfully send an email with a magic link to the user.

## Screenshots

![image](https://github.com/user-attachments/assets/d2978308-1ae9-4909-98df-f84dead3f187)

## System information

- OS: [Linux]
- Browser (if applies) [N/A]
- Version of supabase-js: [N/A]
- Version of Node.js: [v20.11.1]

## Additional context

Basically this line in /internal/api/magic_link.go
`password.Generate(64, 10, 1, false, true)`
Generates an invalid value for this line in  /internal/api/signup.go
`if err := a.checkPasswordStrength(ctx, p.Password); err != nil {`
